src/Network/WebSockets/Connection.hs

```diff
@@ -203,6 +212,9 @@ data Connection = Connection
     -- the first close message but then the other party must respond.  Finally,
     -- the server is in charge of closing the TCP connection.  This IORef tracks
     -- if we have sent a close message and are waiting for the peer to respond.
+    -- needs to be updated to some more generic plugin framework
+    , connectionDeflate   :: !(Message -> IO Message)
+    , connectionInflate   :: !(Message -> IO Message)
```

Comment:
Instead of adding these two methods here, would it be possible to put them into `connectionParse` and `connectionWrite` respectively?  That is, we construct some closure using the inflating code and put that in `connectionWrite` directly.  That way the API is a bit simpler for the user (only two methods to call).